### PR TITLE
fix compilation with nix 2.25

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

**Comment by inclyc** [comment]
> nixt-nightly> ../test/Value.cpp:61:13: error: cannot convert 'NixInt' (aka 'Checked<long long>') to 'long' without a conversion operator

Looks like on macOS (libc++) it is `long long` :cry:

File: libnixt/test/Value.cpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ TEST_F(ValueTest, selectAttrPath) {
   nix::Value &Kern = selectStringViews(*State, Nested, {"c", "d"});
 
   ASSERT_EQ(Kern.type(), nix::ValueType::nInt);
-  ASSERT_EQ(Kern.integer(), 1);
+  ASSERT_EQ(Kern.integer(), nix::NixInt{ 1 });
```
Comment (Mic92) [inline libnixt/test/Value.cpp:61]: This one should be solved now.

File: nixd/lib/Eval/AttrSetProvider.cpp [hunk 2]
```diff
@@ -54,8 +54,8 @@ std::optional<Location> locationOf(nix::PosTable &PTable, nix::Value &V) {
     return std::nullopt;
 
   Position LPos = {
-      .line = static_cast<int>(NixPos.line - 1),
-      .character = static_cast<int>(NixPos.column - 1),
+      .line = static_cast<long int>(NixPos.line - 1),
```
Comment (Mic92) [inline nixd/lib/Eval/AttrSetProvider.cpp:57]: Do you think macos is still fine? `long int` vs `long long`?

Comment (inclyc) [reply nixd/lib/Eval/AttrSetProvider.cpp:57]: Why this type needs to be changed?

File: nixd/lspserver/include/lspserver/Protocol.h [hunk 3]
```diff
@@ -127,12 +127,12 @@ bool fromJSON(const llvm::json::Value &, VersionedTextDocumentIdentifier &,
 
 struct Position {
   /// Line position in a document (zero-based).
-  int line = 0;
+  long int line = 0;
```
Comment (inclyc) [inline nixd/lspserver/include/lspserver/Protocol.h:130]: How about just keep this line as-is?

File: nixd/lib/Eval/AttrSetProvider.cpp (hunk 2 above)
Comment (Mic92) [reply nixd/lib/Eval/AttrSetProvider.cpp:57]: Will check again. You are probably right.

File: flake.nix [hunk 4]
```diff
@@ -1,6 +1,9 @@
 {
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    # TODO: switch back once nix 2.25 hits nixpkgs-unstable
```
Comment (inclyc) [inline flake.nix:3]: Already merged? https://nixpk.gs/pr-tracker.html?pr=355350

**Review by inclyc — APPROVED** [review]

File: flake.nix (hunk 4 above)
Comment (Mic92) [reply flake.nix:3]: Yes. Go ahead.

Comment (Mic92) [reply flake.nix]: done.

**Review by inclyc — APPROVED** [review]

**Pull request merged**